### DISCUSSION TITLE
printing a VectorArrayCoefficient

### DISCUSSION THREAD

**Discussion by gokhalen** [mfem/mfem#3477] [General]
Is there a way to print out what is in `f`  after line 152 has been executed?

https://github.com/mfem/mfem/blob/ff7ed2efc62ab3302329be522b27b8b8b03ee4be/examples/ex2.cpp#L152

**Comment by tzanio** [comment]
There is no direct way to print a `Coefficient` object, but you can project it on a `GridFunction` and print that.

For example:

```c++
   {
      // Project the coefficient `f` onto the GridFunction `x`
      GridFunction x(fespace);
      x.ProjectCoefficient(f);

      // Visualize `x` with GLVis
      socketstream sol_sock("localhost", 19916);
      sol_sock.precision(8);
      sol_sock << "solution\n" << *mesh << x << flush;
   }
```
